endpoints/openrtb2/video_auction.go

```diff
@@ -0,0 +1,495 @@
+	ao.Request = bidReq
+	ao.Response = response
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	//build simplified response
+	bidResp, err := buildVideoResponse(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+	if bidReq.Test == 1 {
+		bidResp.Ext = response.Ext
+	}
+
+	resp, err := json.Marshal(bidResp)
+	//resp, err := json.Marshal(response)
+	if err != nil {
+		errL := []error{err}
+		handleError(labels, w, errL, ao)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(resp)
+
+}
+
+func handleError(labels pbsmetrics.Labels, w http.ResponseWriter, errL []error, ao analytics.AuctionObject) {
+	labels.RequestStatus = pbsmetrics.RequestStatusErr
+	w.WriteHeader(http.StatusInternalServerError)
+	var errors string
+	for _, er := range errL {
+		errors = fmt.Sprintf("%s %s", errors, er.Error())
+	}
+	fmt.Fprintf(w, "Critical error while running the video endpoint: %v", errors)
+	glog.Errorf("/openrtb2/video Critical error: %v", errors)
+	ao.Status = http.StatusInternalServerError
+	ao.Errors = append(ao.Errors, errL...)
+}
+
+func createImpressions(videoReq *openrtb_ext.BidRequestVideo) (imps []openrtb.Imp, err error) {
+	videoDur := videoReq.PodConfig.DurationRangeSec
+	minDuration, maxDuration := minMax(videoDur)
+	reqExactDur := videoReq.PodConfig.RequireExactDuration
+	videoData := videoReq.Video
+
+	finalImpsArray := make([]openrtb.Imp, 0)
+	for _, pod := range videoReq.PodConfig.Pods {
+
+		//load stored impression
+		storedImpressionId := string(pod.ConfigId)
+		storedImp := loadStoredImp(storedImpressionId)
+
+		numImps := pod.AdPodDurationSec / minDuration
+
+		impDivNumber := numImps / len(videoDur)
+
+		impsArray := make([]openrtb.Imp, numImps)
+		for impInd := range impsArray {
+			newImp := createImpressionTemplate(storedImp, videoData)
+			impsArray[impInd] = newImp
+			if reqExactDur {
+				//floor := int(math.Floor(ind/impDivNumber))
+				durationIndex := impInd / impDivNumber
+				if durationIndex > len(videoDur)-1 {
+					durationIndex = len(videoDur) - 1
+				}
+				impsArray[impInd].Video.MaxDuration = int64(videoDur[durationIndex])
+				impsArray[impInd].Video.MinDuration = int64(videoDur[durationIndex])
+				//fmt.Println(podIndex, "  ", impInd, "duration ", videoDur[durationIndex])
+			} else {
+				impsArray[impInd].Video.MaxDuration = int64(maxDuration)
+			}
+
+			impsArray[impInd].ID = fmt.Sprintf("%d_%d", pod.PodId, impInd)
+		}
+		finalImpsArray = append(finalImpsArray, impsArray...)
+
+	}
+	return finalImpsArray, nil
+}
+
+func createImpressionTemplate(imp openrtb.Imp, video openrtb_ext.SimplifiedVideo) openrtb.Imp {
+	imp.Video = &openrtb.Video{}
+	imp.Video.W = video.W
+	imp.Video.H = video.H
+	imp.Video.Protocols = video.Protocols
+	imp.Video.MIMEs = video.Mimes
+	return imp
+}
+
+func loadStoredImp(storedImpId string) openrtb.Imp {
```

Comment:
You should just grab the stored imp from the same source as the openrtb code grabs the stored imp. That is what the plan is, no need for a stub.

Reply:
Added stored impression loading instead of using stub